code/computational_geometry/src/sphere_tetrahedron_intersection/sphere_tetrahedron_intersection.cpp

```diff
@@ -0,0 +1,1237 @@
+/*
+Analytical calculations for sphere-tetrahedron intersections...
+These calculations use a lot of vector math, so I had to include my vector class.
+Vector class ends at around line 900, and intersection codes start.
+
+ - RedBlight
+*/
+
+#ifndef LU_VECTOR_INCLUDED
+#define LU_VECTOR_INCLUDED
+
+#include <sstream>
+#include <string>
+#include <complex>
+#include <vector>
+#include <iostream>
+
+namespace LUV
+{
+	// LuVector.hpp
+	
+	using namespace std;
+
+	template< size_t N, class T >
+	class LuVector;
+
+	template< class T >
+	using LuVector2 = LuVector< 2, T >;
+
+	template< class T >
+	using LuVector3 = LuVector< 3, T >;
+
+	using LuVector2f = LuVector< 2, float >;
+	using LuVector2d = LuVector< 2, double >;
+	using LuVector2c = LuVector< 2, complex< double > >;
+
+	using LuVector3f = LuVector< 3, float >;
+	using LuVector3d = LuVector< 3, double >;
+	using LuVector3c = LuVector< 3, complex< double > >;
+	
+	// LuVector_Op.hpp
+	
+	//////////////////////////////////////////////////////////////
+	// TEMPLATE LOOP INDEX VARIABLE
+	//////////////////////////////////////////////////////////////
+	template< size_t N >
+	struct LOOP_INDEX
+	{
+
+	};
+
+	//////////////////////////////////////////////////////////////
+	// TEMPLATE LOOP UNROLLING OPERATION FUNCTORS
+	//////////////////////////////////////////////////////////////
+
+	template< class T, class S >
+	struct OP_COPY
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs = rhs;
+		}
+	};
+
+	//////////////////////////////////////////////////////////////
+
+	template< class T, class S >
+	struct OP_ADD
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs += rhs;
+		}
+	};
+
+	template< class T, class S >
+	struct OP_SUB
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs -= rhs;
+		}
+	};
+
+	template< class T, class S >
+	struct OP_MUL
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs *= rhs;
+		}
+	};
+
+	template< class T, class S >
+	struct OP_DIV
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs /= rhs;
+		}
+	};
+
+	//////////////////////////////////////////////////////////////
+
+	template< class T, class S >
+	struct OP_ABS
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs = abs( rhs );
```

Comment:
Please avoid `using namespace std` and fully qualify the name.

Reply:
Sure.